src/snowcli/cli/nativeapp/commands.py

```diff
@@ -1,17 +1,21 @@
 from typing import Optional
+from pathlib import Path
+
 import logging
 import typer
 
-from snowcli.cli.common.decorators import global_options
+from snowcli.cli.common.decorators import global_options, global_options_with_connection
 from snowcli.cli.common.flags import DEFAULT_CONTEXT_SETTINGS
 from snowcli.output.decorators import with_output
+from snowcli.cli.stage.diff import DiffResult, stage_diff
```

Comment:
@sfc-gh-turbaszek 
After reading guidelines which you prepared, I wonder if it is ok to use `diff` logic directly, or should it be wrapped by some function in `StageManager`?